reference-library/TransientStateLibrary.md

```diff
@@ -0,0 +1,122 @@
+# Transient State Library
+
+The `TransientStateLibrary` is a crucial component of Uniswap V4, providing utility functions for managing transient state in the PoolManager contract. This library handles operations related to reserves, delta counts, and locking state, which are essential for the efficient and secure operation of the Uniswap V4 protocol.
+
+## Key Concepts
+
+### Transient Storage
+
+Uniswap V4 uses transient storage to optimize gas costs and improve efficiency. Transient storage, introduced in EIP-1153, is a way to store data that is only needed for the duration of a transaction, without persisting it to the blockchain's state trie. This is achieved using the `TLOAD` and `TSTORE` opcodes.
+
+Common operations that involve transient state include:
+
+- Checking reserves (`getReserves`)
+- Verifying currency deltas (`currencyDelta`)
+- Syncing currency states (`sync`)
+- Settling currency balances (`settle`)
+
```

Comment:
can we add information in this section that instead of custom getters exposing transient storage, PoolManager implements `exttload` (_external tload_), and `TransientStateLibrary` makes calls to `exttload` to access transient storage